tableauserverclient/server/endpoint/endpoint.py

```diff
@@ -15,6 +16,16 @@ def _check_status(server_response):
         if server_response.status_code not in Success_codes:
             raise ServerResponseError.from_response(server_response.content)
 
+    def get_unauthenticated_request(self, url, request_object=None):
```

Comment:
Feels like we should probably have a "_get_request" method and these two just call the underlying one. Unauthenticated can call directly. Normal can create the headers list and pass it in. The rest of the code is identical.